### ISSUE TITLE
Segfault with closure

### ISSUE THREAD

**Issue by ruudboon** [zephir-lang/zephir#1912]
In Phalcon when setting a DI like this
```
$di->set('closure', function () {
    $test = "Test";
    exit();
});
```
It wil cause a segfault when retrieving the closure from the DI.
```
$di->get('closure'); //segfault
```
Removing the variable will prevent the segfault.
```
$di->set('closure2', function () {
    exit();
});
$di->get('closure'); //no segfault
```
I'm trying to isolate this issue without the Phalcon framework but would like to already report this for now.

Zephir: 0.12.2
PHP: 7.2.20

**Comment by Jurigag** [comment]
Does the same happen without exit?

**Comment by ruudboon** [comment]
No only with exit.

**Comment by sergeyklay** [comment]
@danhunsaker Could you please take a look by any chance

**Comment by danhunsaker** [comment]
I'll see what I can find.

**Comment by Jurigag** [comment]
@ruudboon well, next time try to add trace from gdb

**Comment by ruudboon** [comment]
@Jurigag I agree, still working on that. Need to find a way to do this in Nanobox. That's why I wrote still working on isolating this issue.

**Comment by CameronHall** [comment]
This was fixed in either 0.12.3 or 0.12.4.

**Comment by ruudboon** [comment]
Confirmed!
